manifests/provider/ldap.pp

```diff
@@ -27,73 +27,74 @@
+  String              $ldap_schema          = 'rfc2307',
+  String              $ldap_default_bind_dn = simplib::lookup('simp_options::ldap::bind_dn', { 'default_value' => "cn=hostAuth,ou=Hosts,%{hiera('simp_options::ldap::base_dn')}" }),
   $ldap_default_authtok_type = '',
-  $ldap_default_authtok = hiera('ldap::bind_pw',''),
-  $ldap_user_object_class = '',
-  $ldap_user_name = '',
-  $ldap_user_uid_number = '',
-  $ldap_user_gid_number = '',
-  $ldap_user_gecos = '',
-  $ldap_user_home_directory = '',
-  $ldap_user_shell = '',
-  $ldap_user_uuid = '',
-  $ldap_user_objectsid = '',
-  $ldap_user_modify_timestamp = '',
-  $ldap_user_shadow_last_change = '',
-  $ldap_user_shadow_min = '',
-  $ldap_user_shadow_max = '',
-  $ldap_user_shadow_warning = '',
-  $ldap_user_shadow_inactive = '',
-  $ldap_user_shadow_expire = '',
-  $ldap_user_krb_last_pwd_change = '',
-  $ldap_user_krb_password_expiration = '',
-  $ldap_user_ad_account_expires = '',
-  $ldap_user_ad_user_account_control = '',
-  $ldap_ns_account_lock = '',
-  $ldap_user_nds_login_disabled = '',
-  $ldap_user_nds_login_expiration_time = '',
-  $ldap_user_nds_login_allowed_time_map = '',
-  $ldap_user_principal = '',
-  $ldap_user_extra_attrs = [],
-  $ldap_user_ssh_public_key = '',
+  String              $ldap_default_authtok = simplib::lookup('simp_options::ldap::bind_hash',{ 'default_value' => '' }),
+  String              $ldap_user_object_class = '',
+  String              $ldap_user_name = '',
+  String              $ldap_user_uid_number = '',
+  String              $ldap_user_gid_number = '',
+  String              $ldap_user_gecos = '',
+  String              $ldap_user_home_directory = '',
+  String              $ldap_user_shell = '',
+  String              $ldap_user_uuid = '',
+  String              $ldap_user_objectsid = '',
+  String              $ldap_user_modify_timestamp = '',
+  String              $ldap_user_shadow_last_change = '',
+  String              $ldap_user_shadow_min = '',
+  String              $ldap_user_shadow_max = '',
+  String              $ldap_user_shadow_warning = '',
+  String              $ldap_user_shadow_inactive = '',
+  String              $ldap_user_shadow_expire = '',
+  String              $ldap_user_krb_last_pwd_change = '',
+  String              $ldap_user_krb_password_expiration = '',
+  String              $ldap_user_ad_account_expires = '',
+  String              $ldap_user_ad_user_account_control = '',
+  String              $ldap_ns_account_lock = '',
+  String              $ldap_user_nds_login_disabled = '',
+  String              $ldap_user_nds_login_expiration_time = '',
+  String              $ldap_user_nds_login_allowed_time_map = '',
+  String              $ldap_user_principal = '',
+  Array              $ldap_user_extra_attrs = [],
+  String              $ldap_user_ssh_public_key = '',
   $ldap_force_upper_case_realm = '',
```

Comment:
It is not strong typed completely, like I said in the commit I just commented out validates and did some minor stuff.  Don't think it worth aligning now. can do that when we strong type.